libraries/rush-lib/src/logic/operations/AsyncOperationQueue.ts

```diff
@@ -166,9 +155,11 @@ export class AsyncOperationQueue
     if (waitingIterators.length > 0) {
       // returns an unassigned operation to let caller decide when there is at least one
       // remote executing operation which is not ready to process.
-      if (queue.some((operation) => operation.status === OperationStatus.RemoteExecuting)) {
+      const remoteExecutingOperation: OperationExecutionRecord | undefined =
+        this.tryGetRemoteExecutingOperation();
+      if (remoteExecutingOperation) {
         waitingIterators.shift()!({
-          value: { weight: 1, status: UNASSIGNED_OPERATION },
+          value: remoteExecutingOperation,
           done: false
         });
       }
```

Comment:
This code path can go away in favor of just returning operations in state `OperationStatus.RemoteExecutingPossiblyComplete` as normal operations in the primary loop. If we don't want them to bounce through the `Queued` status we can add a check and not override before returning.